stardao-mongodb/src/main/java/io/stardog/stardao/mongodb/AbstractMongoDao.java

```diff
@@ -93,6 +93,22 @@ protected Object generateId() {
     }
 
     /**
+     * Load a partial object that might not exist by id.
+     * @param id    primary key value
+     * @param fields    set of fields to return
+     * @return
+     */
+    public Optional<P> loadOpt(K id, Iterable<String> fields) {
```

Comment:
I guess, you can have this method as extra overloaded one but, I think projection can be applied in `loadOpt(K id)` also because `getFieldData()` already has fields metadata. I made a quick hack and added a method in `AbstractMongoDao`:

``` java
    protected Map<String, Object> storageFieldsProjection() {
        ImmutableMap.Builder<String, Object> builder = new ImmutableMap.Builder<>();
        for (final Field field : getFieldData().getMap().values()) {
            builder.put(field.getStorageName(), 1);
        }
        return builder.build();
    }

    @Override
    public Optional<M> loadOpt(K id) {
        Document query = new Document(ID_FIELD, id);
        Document projection = new Document(storageFieldsProjection());
        Document doc = collection.find(query).projection(projection).limit(1).first();
        return Optional.ofNullable(modelMapper.toObject(doc));
    }
```